src/core/client/stream/tabs/Comments/GiphyInput/GiphyInput.tsx

```diff
@@ -1,107 +1,71 @@
 import { Localized } from "@fluent/react/compat";
+import { GiphyFetch, SearchOptions } from "@giphy/js-fetch-api";
+import { IGif } from "@giphy/js-types";
+import { Grid } from "@giphy/react-components";
 import React, {
   ChangeEvent,
   FunctionComponent,
   KeyboardEvent,
   Ref,
   useCallback,
   useEffect,
+  useMemo,
   useRef,
   useState,
 } from "react";
+import useDebounce from "react-use/lib/useDebounce";
+import useResizeObserver from "use-resize-observer";
 
-import { GiphyGif } from "coral-common/types/giphy";
-import { useFetch } from "coral-framework/lib/relay";
 import {
-  BaseButton,
   Button,
   ButtonIcon,
-  Flex,
   HorizontalGutter,
-  Icon,
   InputLabel,
   TextField,
 } from "coral-ui/components/v2";
-import { CallOut } from "coral-ui/components/v3";
 
-import { GIF_RESULTS_LIMIT, GifSearchFetch } from "./GifSearchFetch";
 import GiphyAttribution from "./GiphyAttribution";
 
 import styles from "./GiphyInput.css";
 
+const APPROX_COL_WIDTH = 120;
+
 interface Props {
-  onSelect: (gif: GiphyGif) => void;
+  onSelect: (gif: IGif) => void;
   forwardRef?: Ref<HTMLInputElement>;
+  apiKey: string;
+  maxRating: string;
 }
 
-const GiphyInput: FunctionComponent<Props> = ({ onSelect }) => {
-  const gifSearchFetch = useFetch(GifSearchFetch);
-  const [results, setResults] = useState<GiphyGif[]>([]);
-  const [error, setError] = useState<string | null>(null);
-  const [page, setPage] = useState(0);
+const GiphyInput: FunctionComponent<Props> = ({
+  onSelect,
+  apiKey,
+  maxRating,
+}) => {
   const [query, setQuery] = useState<string>("");
-  const [hasNextPage, setHasNextPage] = useState(false);
-  const [isLoading, setIsLoading] = useState(false);
-  const ref = useRef<HTMLInputElement>(null);
-
+  const [debouncedInput, setDebouncedInput] = useState<string>("");
+  useDebounce(() => setQuery(debouncedInput), 500, [debouncedInput]);
+
+  const inputRef = useRef<HTMLInputElement>(null);
+  const { ref, width = 1 } = useResizeObserver<HTMLDivElement>();
+  const gf = new GiphyFetch(apiKey);
+  const fetchGifs = async (offset: number) =>
+    gf.search(query, {
+      offset,
+      limit: 10,
+      rating: maxRating as SearchOptions["rating"],
+      sort: "relevant",
+    });
```

Comment:
This should be wrapped in a `useMemo`:

```suggestion
  const fetchGifs = useMemo(() => {
    const gf = new GiphyFetch(apiKey);
    return async (offset: number) =>
      gf.search(query, {
        offset,
        limit: 10,
        rating: maxRating as SearchOptions["rating"],
        sort: "relevant",
      });
    }, [apiKey, maxRating]);
```